osu.Server.Queues.ScoreStatisticsProcessor/Processors/UserTotalPerformanceProcessor.cs

```diff
@@ -68,96 +62,53 @@ public void ApplyGlobal(SoloScoreInfo score, MySqlConnection conn)
         public async Task UpdateUserStatsAsync(UserStats userStats, int rulesetId, MySqlConnection connection, MySqlTransaction? transaction = null)
         {
             var dbInfo = LegacyDatabaseHelper.GetRulesetSpecifics(rulesetId);
-            long currentTimestamp = DateTimeOffset.Now.ToUnixTimeSeconds();
-            Ruleset ruleset = LegacyRulesetHelper.GetRulesetFromLegacyId(rulesetId);
-
-            if (beatmapStore == null || buildStore == null || currentTimestamp - lastStoreRefresh > 60)
-            {
-                beatmapStore = await BeatmapStore.CreateAsync(connection, transaction);
-                buildStore = await BuildStore.CreateAsync(connection, transaction);
-                lastStoreRefresh = currentTimestamp;
-            }
 
             List<SoloScore> scores = (await connection.QueryAsync<SoloScore>(
-                "SELECT * FROM scores WHERE `user_id` = @UserId AND `ruleset_id` = @RulesetId AND `preserve` = 1", new
+                "SELECT beatmap_id, pp, accuracy FROM scores s "
+                + "JOIN osu_beatmaps USING (beatmap_id) WHERE "
+                // TODO: This adds a small amount of overhead to the query.
+                // Replace this with a cached lookup once we have a good system in place.
+                + "`approved` in (1,2) AND "
```

Comment:
Is this still relevant? Weren't scores on other beatmaps going to have `ranked = 0` set?

Although as far as I recall we only talked about doing that for qualified, but there are also other cases like loved that were _not_ talked about explicitly...

Reply:
it's not. i think this branch may be outdated me as i have been using my other combined branch

Reply:
I've pushed my off-branch changes to this branch for reference.

Reply:
This is also set in the score performance processor now, and updates the `ranked` flag.